app/src/main/java/com/cornellappdev/uplift/ui/components/goalsetting/ScrollWheel.kt

```diff
@@ -0,0 +1,242 @@
+    selectedTextColor: Color,
+    onItemSelected: (index: Int, item: T) -> Unit = { _, _ -> },
+) {
+    val itemHalfHeight = LocalDensity.current.run { itemHeight.toPx() / 2f }
+    val scrollState = rememberLazyListState(0)
+    var lastSelectedIndex by remember {
+        mutableIntStateOf(0)
+    }
+    var itemsState by remember {
+        mutableStateOf(items)
+    }
+    LaunchedEffect(items) {
+        var targetIndex = items.indexOf(initialItem) - 1
+        targetIndex += ((Int.MAX_VALUE / 2) / items.size) * items.size
+        itemsState = items
+        lastSelectedIndex = targetIndex
+        scrollState.scrollToItem(targetIndex)
+    }
+    LazyColumn(
+        modifier = Modifier
+            .width(width)
+            .height(itemHeight * numberOfDisplayedItems)
+            .clip(RoundedCornerShape(8.dp))
+            .border(1.dp, GRAY01, RoundedCornerShape(8.dp)),
+        state = scrollState,
+        flingBehavior = rememberSnapFlingBehavior(
+            lazyListState = scrollState
+        )
+    ) {
+        items(count = Int.MAX_VALUE, itemContent = { i ->
+            val item = itemsState[i % itemsState.size]
+            Box(
+                modifier = Modifier
+                    .height(itemHeight)
+                    .fillMaxWidth()
+                    .onGloballyPositioned { coordinates ->
+                        val y = coordinates.positionInParent().y - itemHalfHeight
+                        val parentHalfHeight = (itemHalfHeight * numberOfDisplayedItems)
+                        val isSelected =
+                            (y > parentHalfHeight - itemHalfHeight &&
+                                    y < parentHalfHeight + itemHalfHeight)
+                        val index = i - 1
+                        if (isSelected && lastSelectedIndex != index) {
+                            onItemSelected(index % itemsState.size, item)
+                            lastSelectedIndex = index
+                        }
+                    }, contentAlignment = Alignment.Center
+            ) {
+                Text(
+                    text = item.toString(),
+                    style = textStyle,
+                    fontFamily = montserratFamily,
+                    fontWeight = FontWeight.SemiBold,
+                    color = if (lastSelectedIndex == i) {
+                        selectedTextColor
+                    } else {
+                        textColor
+                    },
+                    fontSize = if (lastSelectedIndex == i) {
+                        textStyle.fontSize * itemScaleFact
+                    } else {
+                        textStyle.fontSize
+                    }
+                )
+            }
+        })
+    }
+}
+
+/**
+ * @param initialIsAm: whether the initial selection is AM
+ * @param onValueChanged: callback for when the selection changes
+ * @return AmPmSelector composable that allows the user to select AM or PM
+ */
+@Composable
+fun AmPmSelector(
```

Comment:
Is this to be used anywhere else? Might be best to be put in a separate file or make private if not.